packages/backend/src/open_payments/payment/incoming/service.ts

```diff
@@ -129,7 +140,8 @@ async function createIncomingPayment(
     client,
     expiresAt,
     incomingAmount,
-    metadata
+    metadata,
+    tenantId
```

Comment:
We should add the `tenantId` in `walletAddressService.get` so we make sure its the same tenant for the wallet address and the incoming payment